zenoh/src/net/routing/hat/client/mod.rs

```diff
@@ -0,0 +1,319 @@
+            undeclare_client_queryable(&mut wtables, &mut face_clone, &mut res);
+
+            if res.context.is_some() {
+                for match_ in &res.context().matches {
+                    let mut match_ = match_.upgrade().unwrap();
+                    if !Arc::ptr_eq(&match_, &res) {
+                        get_mut_unchecked(&mut match_)
+                            .context_mut()
+                            .disable_query_routes();
+                        qabls_matches.push(match_);
+                    }
+                }
+                get_mut_unchecked(&mut res)
+                    .context_mut()
+                    .disable_query_routes();
+                qabls_matches.push(res);
+            }
+        }
+        drop(wtables);
+
+        let mut matches_data_routes = vec![];
+        let mut matches_query_routes = vec![];
+        let rtables = zread!(tables.tables);
+        for _match in subs_matches.drain(..) {
+            let mut expr = RoutingExpr::new(&_match, "");
+            matches_data_routes.push((
+                _match.clone(),
+                compute_data_routes(&rtables, &mut expr),
+                compute_matching_pulls(&rtables, &mut expr),
+            ));
+        }
+        for _match in qabls_matches.drain(..) {
+            matches_query_routes.push((_match.clone(), compute_query_routes(&rtables, &_match)));
+        }
+        drop(rtables);
+
+        let mut wtables = zwrite!(tables.tables);
+        for (mut res, data_routes, matching_pulls) in matches_data_routes {
+            get_mut_unchecked(&mut res)
+                .context_mut()
+                .update_data_routes(data_routes);
+            get_mut_unchecked(&mut res)
+                .context_mut()
+                .update_matching_pulls(matching_pulls);
+            Resource::clean(&mut res);
+        }
+        for (mut res, query_routes) in matches_query_routes {
+            get_mut_unchecked(&mut res)
+                .context_mut()
+                .update_query_routes(query_routes);
+            Resource::clean(&mut res);
+        }
+        wtables.faces.remove(&face.id);
+        drop(wtables);
+    }
+
+    fn handle_oam(
+        &self,
+        _tables: &mut Tables,
+        _tables_ref: &Arc<TablesLock>,
+        _oam: Oam,
+        _transport: &TransportUnicast,
+    ) -> ZResult<()> {
+        Ok(())
+    }
+
+    #[inline]
+    fn map_routing_context(
+        &self,
+        _tables: &Tables,
+        _face: &FaceState,
+        _routing_context: NodeId,
+    ) -> NodeId {
+        0
+    }
+
+    fn closing(
+        &self,
+        _tables: &mut Tables,
+        _tables_ref: &Arc<TablesLock>,
+        _transport: &TransportUnicast,
+    ) -> ZResult<()> {
+        Ok(())
+    }
+
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    #[inline]
+    fn ingress_filter(&self, _tables: &Tables, _face: &FaceState, _expr: &mut RoutingExpr) -> bool {
+        true
```

Comment:
Does this mean that `ingress filters` are not available in `client mode` since this functions directly return `true`?

Reply:
To clarify, those filters have nothing to do with interceptors. Those are filtering points for the routing hats to filter out push/request messages before and after the tables lookup. This is pure routing logic.
So this simply means that the client routing algorithm has nothing to filter out before table lookup.